CMakeLists.txt

```diff
@@ -299,10 +299,10 @@ find_package(Eigen 3.1 REQUIRED)
 include_directories(SYSTEM ${EIGEN_INCLUDE_DIRS})
 
 # FLANN (required)
-if(NOT PCL_SHARED_LIBS OR ((WIN32 AND NOT MINGW) AND NOT PCL_BUILD_WITH_FLANN_DYNAMIC_LINKING_WIN32))
-  set(FLANN_USE_STATIC ON)
-endif()
 find_package(FLANN 1.7.0 REQUIRED)
+if(NOT (${FLANN_LIBRARY_TYPE} MATCHES ${PCL_FLANN_REQUIRED_TYPE}) AND (NOT ${PCL_FLANN_REQUIRED_TYPE} MATCHES "DONTCARE"))
```

Comment:
The parentheses are confusing. Please add more to be symmetrical wrt `NOT`

Currently looks like: `NOT (FLANN == PCL) && (NOT PCL == DONTCARE)`

Reply:
Should I just remove the parentheses, like:

 `NOT FLANN == PCL && NOT PCL == DONTCARE` 

OR should it be like `NOT (FLANN == PCL) && NOT (PCL == DONTCARE)` This works as well.

Reply:
More is better :)

Reply:
Went for the last option then :)